services/tasks-svc/internal/task/commands/v1/update_task.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"hwes"
+	"tasks-svc/internal/task/aggregate"
+)
+
+type UpdateTaskCommand struct {
+	hwes.BaseCommand
+	Name        *string
+	Description *string
+}
+
+func NewUpdateTaskCommand(taskID uuid.UUID, name, description *string) *UpdateTaskCommand {
+	return &UpdateTaskCommand{BaseCommand: hwes.NewBaseCommand(taskID), Name: name, Description: description}
+}
+
+type UpdateTaskCommandHandler interface {
+	Handle(ctx context.Context, cmd *UpdateTaskCommand) error
+}
+
+type updateTaskCommandHandler struct {
+	as hwes.AggregateStore
+}
+
+func NewUpdateTaskCommandHandler(as hwes.AggregateStore) *updateTaskCommandHandler {
+	return &updateTaskCommandHandler{as: as}
+}
+
+func (c *updateTaskCommandHandler) Handle(ctx context.Context, command *UpdateTaskCommand) error {
+	a, err := aggregate.LoadTaskAggregate(ctx, c.as, command.AggregateID)
+	if err != nil {
+		return err
+	}
+
+	if command.Name != nil {
+		if err := a.UpdateName(ctx, a.Task.Name, *command.Name); err != nil {
+			return err
+		}
+	}
+
+	if command.Description != nil {
+		if err := a.UpdateDescription(ctx, a.Task.Description, *command.Description); err != nil {
+			return err
+		}
```

Comment:
Because the attribute `description` is missing on the `CreateTaskCommand` `a.Task.Description` will always be empty here on the first `TaskDescriptionUpdatedEvent` even if the description has been set initially.

Reply:
The description is an empty string `""` when the description gets not specified during creation. See `models.NewTask()`.